include/rocksdb/file_system.h

```diff
@@ -640,6 +645,15 @@ class FileSystem : public Customizable {
                                const IOOptions& options, bool* is_dir,
                                IODebugContext* /*dgb*/) = 0;
 
+  // Poll for completion of read IO. The Poll() method should call the callback
+  // provided to ReadAsync to indicate completion. A return value of
+  // IOStatus::NotSupported() indicates that polling is not supported, and
+  // RocksDB will be informed of IO completions via the callback on another
+  // thread.
+  virtual IOStatus Poll(IOHandle* /*io_handle*/) {
```

Comment:
Why is the "Poll" off of the FileSystem while the ReadAsync is off the RandomAccessFile?  Shouldn't we poll the file?  

What is the Status returned if the read has completed?  

This also should not take an IOHandle* -- perhaps by reference or const reference?

Reply:
The Poll call could end up handling completions for multiple different files, so its not specific to a FSRandomAccessFile instance. There could be multiple outstanding file reads and they can complete in any order.